File: server/helm-tracker/src/main/java/com/harbortek/helm/tracker/service/BaselineService.java
/*
 * Copyright [2025] [Harbortek Corp.]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.harbortek.helm.tracker.service;


import com.harbortek.helm.tracker.entity.document.DocumentHistoryEntity;
import com.harbortek.helm.tracker.vo.baseline.BaselineCompare;
import com.harbortek.helm.tracker.vo.baseline.BaselineVo;
import com.harbortek.helm.tracker.vo.items.TrackerItemHistoryVo;

import java.util.List;

public interface BaselineService {
     BaselineVo createBaseline(BaselineVo baseline);

    void createDocumentVersion(Long documentId, Long projectId);

    BaselineVo updateBaseline(BaselineVo baseline);

     void deleteBaseline(Long id);

     BaselineVo findOneBaseline(Long id);

     List<BaselineVo> findBaselinesByProjectId(Long projectId, String keyword);

     List<BaselineVo> findBaselinesByDocumentId(Long projectId,Long documentId);

    List<BaselineCompare> compareBaseline(Long projectId,Long leftId, Long rightId);

     List<TrackerItemHistoryVo> findItemsHistory(List<Long> historyIds);

    List<DocumentHistoryEntity> findDocumentHistory(List<Long> historyIds);
}

File: web/src/services/baseline/BaselineService.js
import { request, METHOD } from "@/utils/request";
export function findBaselinesByProjectId(projectId, keyword) {
  return request({
    url: "/tracker/baseline/listByProject",
    method: METHOD.GET,
    params: { projectId: projectId, keyword: keyword },
  });
}
export function listByDocument(projectId, documentId) {
  return request({
    url: "/tracker/baseline/listByDocument",
    method: METHOD.GET,
    params: { projectId: projectId, documentId: documentId },
  });
}
export function createBaseline(parameter) {
  return request({
    url: "/tracker/baseline",
    method: METHOD.POST,
    data: parameter,
  });
}

export function updateBaseline(parameter) {
  return request({
    url: "/tracker/baseline",
    method: METHOD.PUT,
    data: parameter,
  });
}
export function deleteBaseline(id) {
  return request({
    url: "/tracker/baseline/" + id,
    method: METHOD.DELETE,
  });
}

export function compareBaseline(projectId, lefrId, rightId) {
  return request({
    url: "/tracker/baseline/compare",
    method: METHOD.GET,
    params: { projectId: projectId, leftId: lefrId, rightId: rightId },
  });
}

export function findItemsHistory(historyIds) {
  return request({
    url: "/tracker/baseline/item-history",
    method: METHOD.POST,
    data: historyIds,
  });
}
export function findDocumentHistory(historyIds) {
  return request({
    url: "/tracker/baseline/document-history",
    method: METHOD.POST,
    data: historyIds,
  });
}
